tools/underwater-gps.py

```diff
@@ -60,6 +68,108 @@
                                       # E=Estimated, N=not valid, S=Simulator.
        + "*")                         # Checksum
 
+
+def report_status(*args):
+    """
+    reports the current status of this service
+    """
+    # Do not report the same status multiple times
+    global last_status
+    if args == last_status:
+        return
+    last_status = args
+    print(" ".join(args))
+    try:
+        requests.post(STATUS_REPORT_URL, data={"waterlinked": " ".join(args)})
+    except:
+        print("Unable to talk to webui! Could not report status")
+
+
+def get_mavlink(path):
+    """
+    Helper to get mavlink data from mavlink2rest
+    Example: get_mavlink('/VFR_HUD')
+    Returns the data as text
+    """
+    try:
+        return urllib2.urlopen(MAVLINK2REST_URL + '/mavlink' + path).read()
+    except:
+        report_status("Error trying to access mavlink2rest!")
+        return "0.0"
+
+
+def get_message_frequency(message_name):
+    """
+    Returns the frequency at which message "message_name" is being received, 0 if unavailable
+    """
+    try:
+        return float(get_mavlink('/{0}/message_information/frequency'.format(message_name)))
```

Comment:
How can this fail if get_mavlink already has the try except ?

Reply:
yep, got catch!